rust/sbp/src/sbp_string.rs

```diff
@@ -37,6 +39,100 @@ where
     }
 }
 
+impl<T: AsRef<[u8]>> SbpString<T, Unterminated> {
+    /// Checked unterminated SbpString builder, verify last byte is non null
+    pub fn unterminated(data: T) -> Result<Self, UnterminatedError> {
+        match data.as_ref().last() {
+            Some(l) if l != &0 => Ok(Self::new(data)),
+            _ => Err(UnterminatedError),
+        }
+    }
+}
+
+impl<T: AsRef<[u8]>> SbpString<T, NullTerminated> {
+    pub fn null_terminated(data: T) -> Result<Self, NullTerminatedError> {
+        match data.as_ref().last() {
+            Some(l) if l == &0 => Ok(Self::new(data)),
+            _ => Err(NullTerminatedError),
+        }
+    }
+}
+
+/// Helper function to alternate null bytes
+/// Used in [`SbpString::from_parts`]
+///
+/// Ex. [a, b, c] => [a, 0, b, 0, c, 0]
+///
+/// Returns null terminated vector
+fn alt_null<R, I>(parts: I) -> Vec<u8>
+where
+    R: AsRef<[u8]>,
+    I: IntoIterator<Item = R>,
+{
+    let mut data = vec![];
+    for i in parts {
+        for j in i.as_ref() {
+            data.push(*j)
+        }
+        data.push(0);
+    }
+    data
+}
+
+// pub type Parts<'a> = std::slice::Split<'a, u8, fn(&u8) -> bool>;
```

Comment:
@notoriaga wasn't sure how to use this type...

maybe we could store the parts in SbpString i.e...?

```rust
#[derive(Debug, PartialEq, Clone)]
pub struct SbpString<T, E = Unterminated> {
    data: T,
    encoding: PhantomData<E>,
    parts: Vec<Cow<str>>
}
```